pinot-core/src/main/java/com/linkedin/pinot/core/query/aggregation/function/DistinctCountHLLAggregationFunction.java

```diff
@@ -180,15 +165,7 @@ public Long reduce(List<HyperLogLog> combinedResultList) {
     if ((combinedResultList == null) || combinedResultList.isEmpty()) {
       return 0L;
     }
-    HyperLogLog reducedResult = combinedResultList.get(0);
-    for (int i = 1; i < combinedResultList.size(); ++i) {
-      try {
-        reducedResult.addAll(combinedResultList.get(i));
-      } catch (CardinalityMergeException e) {
-        LOGGER.error("Caught exception while merging Cardinality using HyperLogLog", e);
-        Utils.rethrowException(e);
-      }
-    }
+    HyperLogLog reducedResult = HllUtil.mergeHLLResultsToFirstInList(combinedResultList);
```

Comment:
Is this shared with fastHLL ? Add a comment to FastHLL class reduce method that reduce is implemented here